singer/transform.py

```diff
@@ -0,0 +1,95 @@
+import datetime
+import pendulum
+from singer import utils
+
+
+def _transform_object(data, prop_schema):
+    return {k: transform(v, prop_schema[k]) for k, v in data.items() if k in prop_schema}
+
+def _transform_array(data, item_schema):
+    return [transform(row, item_schema) for row in data]
+
+def unix_milliseconds_to_datetime(value):
+    return utils.strftime(datetime.datetime.utcfromtimestamp(int(value) * 0.001))
+
+def unix_seconds_to_datetime(value):
+    return utils.strftime(datetime.datetime.utcfromtimestamp(int(value)))
+
+def string_to_datetime(value):
+    return  utils.strftime(pendulum.parse(value))
+
+def _transform_datetime(value, integer_datetime_fmt):
+    if integer_datetime_fmt not in [NO_INTEGER_DATETIME_PARSING,
+                                    UNIX_SECONDS_INTEGER_DATETIME_PARSING,
+                                    UNIX_MILLISECONDS_INTEGER_DATETIME_PARSING]:
+        raise Exception("Invalid integer datetime parsing option")
+
+    if integer_datetime_fmt == NO_INTEGER_DATETIME_PARSING:
+        return string_to_datetime(value)
+    else:
+        try:
+            if integer_datetime_fmt == UNIX_SECONDS_INTEGER_DATETIME_PARSING:
+                return unix_seconds_to_datetime(value)
+            elif integer_datetime_fmt == UNIX_MILLISECONDS_INTEGER_DATETIME_PARSING:
+                return unix_milliseconds_to_datetime(value)
+        except:
+            return string_to_datetime(value)
+
+NO_INTEGER_DATETIME_PARSING = 0
```

Comment:
I think I'd prefer string constants rather than ints. Is there a reason why we couldn't use string constants here?